File: jadx-gui/src/main/java/jadx/gui/plugins/script/ScriptCompletionData.java
package jadx.gui.plugins.script;

import javax.swing.Icon;
import javax.swing.text.JTextComponent;

import org.fife.ui.autocomplete.Completion;
import org.fife.ui.autocomplete.CompletionProvider;

public class ScriptCompletionData implements Completion {

	private final CompletionProvider provider;
	private final int relevance;

	private String input;
	private String code;
	private int replacePos;
	private Icon icon;
	private String summary;
	private String toolTip;

	public ScriptCompletionData(CompletionProvider provider, int relevance) {
		this.provider = provider;
		this.relevance = relevance;
	}

	public void setData(String input, String code, int replacePos) {
		this.input = input;
		this.code = code;
		this.replacePos = replacePos;
	}

	@Override
	public String getInputText() {
		return input;
	}

	@Override
	public CompletionProvider getProvider() {
		return provider;
	}

	@Override
	public String getAlreadyEntered(JTextComponent comp) {
		return provider.getAlreadyEnteredText(comp);
	}

	@Override
	public int getRelevance() {
		return relevance;
	}

	@Override
	public String getReplacementText() {
		return code.substring(0, replacePos) + input;
	}

	@Override
	public Icon getIcon() {
		return icon;
	}

	public void setIcon(Icon icon) {
		this.icon = icon;
	}

	@Override
	public String getSummary() {
		return summary;
	}

	public void setSummary(String summary) {
		this.summary = summary;
	}

	@Override
	public String getToolTipText() {
		return toolTip;
	}

	public void setToolTip(String toolTip) {
		this.toolTip = toolTip;
	}

	@Override
	public int compareTo(Completion other) {
		return Integer.compare(relevance, other.getRelevance());
	}

	@Override
	public String toString() {
		return input;
	}
}

File: jadx-plugins/jadx-script-kotlin/src/main/kotlin/jadx/plugins/script/kotlin/gui/ScriptCompletionData.kt
package jadx.plugins.script.kotlin.gui

import org.fife.ui.autocomplete.Completion
import org.fife.ui.autocomplete.CompletionProvider
import javax.swing.Icon
import javax.swing.text.JTextComponent

class ScriptCompletionData(
	private val provider: CompletionProvider,
	private val relevance: Int,
	private val input: String,
	private val code: String,
	private val replacePos: Int,
	private val icon: Icon,
	private val toolTip: String,
	private val summary: String,
) : Completion {

	override fun getInputText(): String {
		return input
	}

	override fun getProvider(): CompletionProvider {
		return provider
	}

	override fun getAlreadyEntered(comp: JTextComponent?): String? {
		return provider.getAlreadyEnteredText(comp)
	}

	override fun getRelevance(): Int {
		return relevance
	}

	override fun getReplacementText(): String {
		return code.substring(0, replacePos) + input
	}

	override fun getIcon(): Icon {
		return icon
	}

	override fun getSummary(): String {
		return summary
	}

	override fun getToolTipText(): String {
		return toolTip
	}

	override fun compareTo(other: Completion): Int {
		return relevance.compareTo(other.relevance)
	}

	override fun toString(): String {
		return input
	}
}
